### PR TITLE
[ie/bbc] extract more formats

### LINKED ISSUE #4902: Why does YT-DLP only download SD in 396p from iPlayer yet get_iplayer downloads in 540p?
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I remove or skip any mandatory\* field

### Checklist

- [X] I'm asking a question and **not** reporting a bug or requesting a feature
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2022.09.01** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar questions **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Please make sure the question is worded well enough to be understood

Trying to download this standard definition video in the best quality: 
https://www.bbc.co.uk/iplayer/episode/p00yzpg2/top-gear-series-1-episode-1

Get_iplayer downloads it in the best quality it can get at 960 x 540p 25 fps with AAC 96 Kbps
YT-DLP downloading in the best quality only gets it in 704 x 396p 50 fps, however it gets a better audio quality at AAC 128 Kbps. 

YT-DLP --list-formats can't even see the 960 x 540p. Why is that when get_iplayer can?

How can I get yt-dlp to download the 960 x 540p version? Here's my script:

`-f ("bestvideo[width>=1920]"/bestvideo[ext=avc1])+bestaudio/best `

Thanks

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
X:\Portable Installations\YouTube-DL>cd /D "X:\Portable Installations\YouTube-DL
\"

X:\Portable Installations\YouTube-DL>yt-dlp -vU yt-dlp https://www.bbc.co.uk/ipl
ayer/episode/p00yzpg2/top-gear-series-1-episode-1 --list-formats
[debug] Command-line config: ['-vU', 'yt-dlp', 'https://www.bbc.co.uk/iplayer/ep
isode/p00yzpg2/top-gear-series-1-episode-1', '--list-formats']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8 (No VT), erro
r utf-8 (No VT), screen utf-8 (No VT)
[debug] yt-dlp version 2022.09.01 [5d7c7d6] (win32_exe)
[debug] Python 3.8.10 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg N-65018-gad91bf8 (needs_adtstoasc), ffprobe N-65018
-gad91bf8
[debug] Optional libraries: Cryptodome-3.15.0, brotli-1.0.9, certifi-2022.06.15,
 mutagen-1.45.1, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
[debug] Loaded 1670 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp/releas
es/latest
Latest version: 2022.09.01, Current version: 2022.09.01
yt-dlp is up to date (2022.09.01)
[debug] [CommonMistakes] Extracting URL: yt-dlp
ERROR: [CommonMistakes] You've asked yt-dlp to download the URL "yt-dlp". That d
oesn't make any sense. Simply remove the parameter in your command or configurat
ion.
  File "yt_dlp\extractor\common.py", line 670, in extract
  File "yt_dlp\extractor\commonmistakes.py", line 25, in _real_extract

[debug] [bbc.co.uk] Extracting URL: https://www.bbc.co.uk/iplayer/episode/p00yzp
g2/top-gear-series-1-episode-1
[bbc.co.uk] p00yzpg2: Downloading video page
[bbc.co.uk] p00yzpg2: Downloading playlist JSON
[bbc.co.uk] p00yzpg3: Downloading media selection JSON
[bbc.co.uk] p00yzpg3: Downloading m3u8 information
[bbc.co.uk] p00yzpg3: Downloading m3u8 information
[bbc.co.uk] p00yzpg3: Downloading m3u8 information
[bbc.co.uk] p00yzpg3: Downloading m3u8 information
[bbc.co.uk] p00yzpg3: Downloading m3u8 information
[bbc.co.uk] p00yzpg3: Downloading m3u8 information
[bbc.co.uk] p00yzpg3: Downloading MPD manifest
[bbc.co.uk] p00yzpg3: Downloading MPD manifest
[bbc.co.uk] p00yzpg3: Downloading MPD manifest
[bbc.co.uk] p00yzpg3: Downloading MPD manifest
[bbc.co.uk] p00yzpg3: Downloading MPD manifest
[bbc.co.uk] p00yzpg3: Downloading MPD manifest
[debug] Formats sorted by: hasvid, ie_pref, lang, quality, res, fps, hdr:12(7),
vcodec:vp9.2(10), channels, acodec, filesize, fs_approx, tbr, vbr, abr, asr, pro
to, vext, aext, hasaud, source, id
[info] Available formats for p00yzpg3:
ID                                 EXT RESOLUTION FPS |   FILESIZE   TBR PROTO |
 VCODEC        VBR ACODEC      ABR ASR MORE INFO
--------------------------------------------------------------------------------
--------------------------------------------------------------
mf_akamai-audio_eng_1=128000-0     m4a audio only     | ~ 56.25MiB  128k dash  |
 audio only        mp4a.40.2  128k 48k [en] Original, m4a_dash
mf_akamai-audio_eng_1=128000-1     m4a audio only     | ~ 56.25MiB  128k dash  |
 audio only        mp4a.40.2  128k 48k [en] Original, m4a_dash
mf_bidi-audio_eng_1=128000-0       m4a audio only     | ~ 56.25MiB  128k dash  |
 audio only        mp4a.40.2  128k 48k [en] Original, m4a_dash
mf_bidi-audio_eng_1=128000-1       m4a audio only     | ~ 56.25MiB  128k dash  |
 audio only        mp4a.40.2  128k 48k [en] Original, m4a_dash
mf_cloudfront-audio_eng_1=128000-0 m4a audio only     | ~ 56.25MiB  128k dash  |
 audio only        mp4a.40.2  128k 48k [en] Original, m4a_dash
mf_cloudfront-audio_eng_1=128000-1 m4a audio only     | ~ 56.25MiB  128k dash  |
 audio only        mp4a.40.2  128k 48k [en] Original, m4a_dash
mf_akamai-video=827000-0           mp4 704x396     25 | ~363.43MiB  827k dash  |
 avc3.4D401E  827k video only          Original, mp4_dash
mf_akamai-video=827000-1           mp4 704x396     25 | ~363.43MiB  827k dash  |
 avc3.4D401E  827k video only          Original, mp4_dash
mf_bidi-video=827000-0             mp4 704x396     25 | ~363.43MiB  827k dash  |
 avc3.4D401E  827k video only          Original, mp4_dash
mf_bidi-video=827000-1             mp4 704x396     25 | ~363.43MiB  827k dash  |
 avc3.4D401E  827k video only          Original, mp4_dash
mf_cloudfront-video=827000-0       mp4 704x396     25 | ~363.43MiB  827k dash  |
 avc3.4D401E  827k video only          Original, mp4_dash
mf_cloudfront-video=827000-1       mp4 704x396     25 | ~363.43MiB  827k dash  |
 avc3.4D401E  827k video only          Original, mp4_dash
mf_akamai-1013-0                   mp4 704x396     25 | ~445.17MiB 1013k m3u8  |
 avc1.4D401E  827k mp4a.40.2  128k     Original
mf_akamai-1013-1                   mp4 704x396     25 | ~445.17MiB 1013k m3u8  |
 avc1.4D401E  827k mp4a.40.2  128k     Original
mf_bidi-1013-0                     mp4 704x396     25 | ~445.17MiB 1013k m3u8  |
 avc1.4D401E  827k mp4a.40.2  128k     Original
mf_bidi-1013-1                     mp4 704x396     25 | ~445.17MiB 1013k m3u8  |
 avc1.4D401E  827k mp4a.40.2  128k     Original
mf_cloudfront-1013-0               mp4 704x396     25 | ~445.17MiB 1013k m3u8  |
 avc1.4D401E  827k mp4a.40.2  128k     Original
mf_cloudfront-1013-1               mp4 704x396     25 | ~445.17MiB 1013k m3u8  |
 avc1.4D401E  827k mp4a.40.2  128k     Original
mf_akamai-video=1570000-0          mp4 704x396     50 | ~689.94MiB 1570k dash  |
 avc3.64001F 1570k video only          Original, mp4_dash
mf_akamai-video=1570000-1          mp4 704x396     50 | ~689.94MiB 1570k dash  |
 avc3.64001F 1570k video only          Original, mp4_dash
mf_bidi-video=1570000-0            mp4 704x396     50 | ~689.94MiB 1570k dash  |
 avc3.64001F 1570k video only          Original, mp4_dash
mf_bidi-video=1570000-1            mp4 704x396     50 | ~689.94MiB 1570k dash  |
 avc3.64001F 1570k video only          Original, mp4_dash
mf_cloudfront-video=1570000-0      mp4 704x396     50 | ~689.94MiB 1570k dash  |
 avc3.64001F 1570k video only          Original, mp4_dash
mf_cloudfront-video=1570000-1      mp4 704x396     50 | ~689.94MiB 1570k dash  |
 avc3.64001F 1570k video only          Original, mp4_dash
mf_akamai-1800-0                   mp4 704x396     50 | ~791.02MiB 1800k m3u8  |
 avc1.64001F 1570k mp4a.40.2  128k     Original
mf_akamai-1800-1                   mp4 704x396     50 | ~791.02MiB 1800k m3u8  |
 avc1.64001F 1570k mp4a.40.2  128k     Original
mf_bidi-1800-0                     mp4 704x396     50 | ~791.02MiB 1800k m3u8  |
 avc1.64001F 1570k mp4a.40.2  128k     Original
mf_bidi-1800-1                     mp4 704x396     50 | ~791.02MiB 1800k m3u8  |
 avc1.64001F 1570k mp4a.40.2  128k     Original
mf_cloudfront-1800-0               mp4 704x396     50 | ~791.02MiB 1800k m3u8  |
 avc1.64001F 1570k mp4a.40.2  128k     Original
mf_cloudfront-1800-1               mp4 704x396     50 | ~791.02MiB 1800k m3u8  |
 avc1.64001F 1570k mp4a.40.2  128k     Original

X:\Portable Installations\YouTube-DL>PAUSE
Press any key to continue . . .
```

### PR DESCRIPTION
**Pull request by barsnick**
This fetches the higher-resolution SD streams.

Taken from
https://github.com/yt-dlp/yt-dlp/issues/4902#issuecomment-1244298806 by @dirkf.

Fixes https://github.com/yt-dlp/yt-dlp/issues/4902

**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

This adds additional parsing of formats, to deliver the higher-resolution, and also higher-quality, SD streams. Notably, this discovers the 540p streams, over the previous maximum resolution of 396p.

The "Author" tag of the commit credits @dirkf, who provided the diff in the named comment.

Some of the links from the ticket were tested manually, both for finding the formats and for actual download of the media.

The IEs' tests should not be invalidated, as none of them use content MD5 sums. Many tests fail for me anyway, they didn't seem of much use. (Requires clean-up.)

Fixes #4902

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [ ] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [x] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details><details><summary>Copilot Summary</summary>  

### Summary
🌐🎞️⚠️

Enhance `bbc.py` extractor to handle media set failures more gracefully and collect more formats and subtitles.

> _When the BBC extractor runs_
> _It may encounter media shuns_
> _But instead of an error_
> _It shows some more terror_
> _By collecting all formats and `subs`_

### Walkthrough
* Collect and return all formats and subtitles from different media sets in `_download_media_selector` ([link](https://github.com/yt-dlp/yt-dlp/pull/8321/files?diff=unified&w=0#diff-0abdd53ffd6c2083b41d6d68bf14219a98fb7f4dcf0199af518592fa89439f57L320-R327), [link](https://github.com/yt-dlp/yt-dlp/pull/8321/files?diff=unified&w=0#diff-0abdd53ffd6c2083b41d6d68bf14219a98fb7f4dcf0199af518592fa89439f57L329-R339))

</details>

### CODE REVIEW

**Comment by barsnick** [comment]
Force-pushed only to fix a typo in the commit message. I dared to do so, as no reviews were commented yet.

File: yt_dlp/extractor/bbc.py [hunk 1]
```diff
@@ -317,16 +317,26 @@ def _raise_extractor_error(self, media_selection_error):
 
     def _download_media_selector(self, programme_id):
         last_exception = None
+        formats, subtitles = [], {}
         for media_set in self._MEDIA_SETS:
             try:
-                return self._download_media_selector_url(
+                fmts, sttls = self._download_media_selector_url(
```
Comment (bashonly) [inline yt_dlp/extractor/bbc.py:323]: `sttls` is less obvious imo
```suggestion
                fmts, subs = self._download_media_selector_url(
```

File: yt_dlp/extractor/bbc.py [hunk 2]
```diff
@@ -317,16 +317,26 @@ def _raise_extractor_error(self, media_selection_error):
...
                     self._MEDIA_SELECTOR_URL_TEMPL % (media_set, programme_id), programme_id)
+                formats.extend(fmts)
+                if sttls:
+                    subtitles = self._merge_subtitles(subtitles, sttls)
```
Comment (bashonly) [inline yt_dlp/extractor/bbc.py:327]:
```suggestion
                self._merge_subtitles(subs, target=subtitles)
```

File: yt_dlp/extractor/bbc.py [hunk 3]
```diff
@@ -317,16 +317,26 @@ def _raise_extractor_error(self, media_selection_error):
...
             except BBCCoUkIE.MediaSelectionError as e:
                 if e.id in ('notukerror', 'geolocation', 'selectionunavailable'):
                     last_exception = e
                     continue
                 self._raise_extractor_error(e)
-        self._raise_extractor_error(last_exception)
+        if last_exception:
+            if formats or subtitles:
+                self.report_warning(
+                    '%s returned error: %s' % (self.IE_NAME, last_exception.id))
```
Comment (bashonly) [inline yt_dlp/extractor/bbc.py:336]: f-string
```suggestion
                self.report_warning(f'{self.IE_NAME} returned error: {last_exception.id}')
```

**Review by bashonly — CHANGES_REQUESTED** [review]
sorry it took so long to review, just some nitpicks to improve readability

**Comment by barsnick** [comment]
I will modify this. Thanks.

File: yt_dlp/extractor/bbc.py (hunk 1 above)
Comment (barsnick) [reply yt_dlp/extractor/bbc.py:323]: Agree.

(This was directly from the patch proposal as referenced in the commit message.)

File: yt_dlp/extractor/bbc.py (hunk 2 above)
Comment (barsnick) [reply yt_dlp/extractor/bbc.py:327]:
```
self._merge_subtitles(subs, target=subtitles)
```
`_merge_subtitles`  does not accept a non-`None` argument.

**Review by bashonly — APPROVED** [review]

**Pull request merged**